### PR TITLE
[T.F 2.0 API Docs] Modifying python math_ops.py documentation

### PR DESCRIPTION
**Pull request by Harsh188**
PR for issue #25802.

Modifying the following existing functions:
-  `math.subtract`

Adding new functions:
- `math.acos`
- `math.add`
- `math.floor`

### CODE REVIEW

**Review by hamelsmu — DISMISSED** [review]

**Comment by hamelsmu** [comment]
Sorry about the random approval, that was a mistake on my part.  Please ignore

File: tensorflow/python/ops/math_ops.py [hunk 1]
```diff
@@ -533,6 +533,30 @@ def _mul(x, y, name=None):
...
 def subtract(x, y, name=None):
+  """Returns x - y element-wise.
+
+  *Note*: Subtract supports broadcasting. More about broadcasting 
+  [here](https://numpy.org/doc/stable/user/basics.broadcasting.html)
+  
+  For example:
+  Both input and output have a range `(-inf, inf)`.
+
+  # `x` and `y` are tensors of shape (1)
```
Comment (mihaimaruseac) [inline tensorflow/python/ops/math_ops.py:544]:
```suggestion
  # `x` and `y` are tensors of shape `(5,)`
```

File: tensorflow/python/ops/math_ops.py [hunk 2]
```diff
@@ -4927,3 +4951,91 @@ def rsqrt(x, name=None):
...
+
+  *NOTE*: `Add` supports broadcasting. `AddN` does not. More about broadcasting
+  [here](http://docs.scipy.org/doc/numpy/user/basics.broadcasting.html)
+
+  Given two input tensors, the `tf.add` operation computes the sum for every element in the tensor.
+
+  For example:
+  Both input and output have a range `(-inf, inf)`.
+
+  # `x` and `y` are tensors of shape (1)
```
Comment (mihaimaruseac) [inline tensorflow/python/ops/math_ops.py:4970]:
```suggestion
  # `x` and `y` are tensors of shape `(5,)`
```

File: tensorflow/python/ops/math_ops.py [hunk 3]
```diff
@@ -533,6 +533,30 @@ def _mul(x, y, name=None):
 @tf_export("math.subtract", "subtract")
 @dispatch.add_dispatch_support
 def subtract(x, y, name=None):
+  """Returns x - y element-wise.
+
+  *Note*: Subtract supports broadcasting. More about broadcasting 
+  [here](https://numpy.org/doc/stable/user/basics.broadcasting.html)
+  
+  For example:
+  Both input and output have a range `(-inf, inf)`.
```
Comment (mihaimaruseac) [inline tensorflow/python/ops/math_ops.py:542]:
```suggestion
  Both input and output have a range `(-inf, inf)`.

  For example:
```

File: tensorflow/python/ops/math_ops.py [hunk 4]
```diff
@@ -4927,3 +4951,91 @@ def rsqrt(x, name=None):
...
+def add(x,y,name=None):
+  """Returns x + y element-wise.
+
+  *NOTE*: `Add` supports broadcasting. `AddN` does not. More about broadcasting
+  [here](http://docs.scipy.org/doc/numpy/user/basics.broadcasting.html)
+
+  Given two input tensors, the `tf.add` operation computes the sum for every element in the tensor.
+
+  For example:
+  Both input and output have a range `(-inf, inf)`.
```
Comment (mihaimaruseac) [inline tensorflow/python/ops/math_ops.py:4968]:
```suggestion
  Both input and output have a range `(-inf, inf)`.

  For example:
```

File: tensorflow/python/ops/math_ops.py [hunk 5]
```diff
@@ -4927,3 +4951,91 @@ def rsqrt(x, name=None):
...
+  """Computes acos of x element-wise.
+
+  Provided an input tensor, the `tf.math.acos` operation returns the inverse cosine of 
+  each element of the tensor. If `y = tf.math.cos(x)` then, `x = tf.math.acos(y)`. 
+
+  Input range is `[-1, 1]` and the output has a range of `[0, pi]`.
+
+  For example:
+
+  # `x` is a tensor of the shape(1)
```
Comment (mihaimaruseac) [inline tensorflow/python/ops/math_ops.py:5002]:
```suggestion
  # `x` is a tensor of the shape `(6,)`
```

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]
I would actually remove the `# ... tensors have shape ..` lines in the example as they can create confusion w.r.t. allowed input shapes.

**Comment by Harsh188** [comment]
Just finished making all the requested changes. I took your advice and removed the `# ... tensors have shape ..` to avoid confusion.

**Review by mihaimaruseac — DISMISSED** [review]

**Comment by rthadur** [comment]
@Harsh188 can you please check ubuntu sanity build failures ?

**Comment by Harsh188** [comment]
I'm not too sure if fixing the pylint issues will fix the Ubuntu Sanity build failure. I went through the logs and saw that they were causing an error so that's what I ended up fixing. I'm still new contributing so, please forgive me for my small mistakes :)

**Review by mihaimaruseac — DISMISSED** [review]

**Review by mihaimaruseac — DISMISSED** [review]

**Review by rthadur — DISMISSED** [review]

**Comment by Harsh188** [comment]
I still can't figure out why I'm getting these build failures and if they are fatal or if I should just ignore them. The build that's failing is `tensorflow/tools/ci_build/builds:gen_win_out` but I'm not sure how that relates to the files that I have made changes to. 

@rthadur can you help me sort this out?

**Comment by rthadur** [comment]
I am not sure about the errors , @mihaimaruseac can you please help ?

**Comment by mihaimaruseac** [comment]
Looks like an import failure. Let's try again and if this still does not work let's try rebasing back on master. If that still does not work, we might need to manually import

**Comment by Harsh188** [comment]
@mihaimaruseac I rebased to master.

**Review by rthadur — DISMISSED** [review]

**Comment by Harsh188** [comment]
> Looks like an import failure. Let's try again and if this still does not work let's try rebasing back on master. If that still does not work, we might need to manually import

@mihaimaruseac how should I go about and manually import? It doesn't look like rebasing back on master has been effective.

**Comment by mihaimaruseac** [comment]
I am going to import it manually, there are some steps that need to be done from inside Google. Sorry for the delay.

**Review by mihaimaruseac — APPROVED** [review]

**Comment by mihaimaruseac** [comment]
Found the issue. You also needed to change `tensorflow/core/api_def/python_api/api_def_Add.pbtxt` and for the other ops you add the entire glue code in Python (see the example for `Sub`)

I did the changes manually on import, so no need to do anything here now. It is pending review now, though likely this would be Wednesday.

**Comment by Harsh188** [comment]
> Found the issue. You also needed to change `tensorflow/core/api_def/python_api/api_def_Add.pbtxt` and for the other ops you add the entire glue code in Python (see the example for `Sub`)
> 
> I did the changes manually on import, so no need to do anything here now. It is pending review now, though likely this would be Wednesday.

@mihaimaruseac awesome!

Regarding the changes in `tensorflow/core/api_def/python_api/api_def_Add.pbtxt` I'm assuming I have to remove the endpoints and add `visibility: HIDDEN`? That's what the other .pbtxt files seem to do but I'm not too sure.

Is glue code the [`@deprecation.deprecated`](https://github.com/tensorflow/tensorflow/blob/712fd5cfe4a5de420a4226c874664423eba4cb1a/tensorflow/python/ops/math_ops.py#L557) section in `sub`?

I'm new to contributing to open source and to the field in general, I apologize for the trouble.

**Comment by mihaimaruseac** [comment]
Sorry again for the delayed response.

By glue code I meant the python function which has the docstring and just returns from `gen_*_ops`.

The dispatch decorator seems to be causing another test to fail, I'll try to debug by end of the week

**Comment by Harsh188** [comment]
> By glue code I meant the python function which has the docstring and just returns from `gen_*_ops`.
> 
> The dispatch decorator seems to be causing another test to fail, I'll try to debug by end of the week

Great, thank you! I could definitely learn quite a lot from you :)

**Comment by mihaimaruseac** [comment]
Unfortunately, due to the way `add` is used in dispatch (something internal, not yet exported externally), I had to remove that from the PR, during manual import.

Once dispatch is fixed, we'll add back documentation for add.

**Pull request merged**